src/pages/ProductDetail.jsx

```diff
@@ -0,0 +1,244 @@
+import ReturnIcon from "../assets/icons/icon_return";
+import noComment from "../assets/images/no-comments.png";
+
+export default function ProductDetail() {
+  const [item, setItem] = useState(null);
+  const [comments, setComments] = useState([]);
+  const [openMenuId, setOpenMenuId] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [editingCommentId, setEditingCommentId] = useState(null);
+  const [editedContent, setEditedContent] = useState("");
+  const [commentText, setCommentText] = useState("");
+  const { productId } = useParams();
+
+  const getDetailData = async () => {
+    try {
+      setIsLoading(true);
+      const data = await getItemDetail(productId);
+      setItem(data);
+    } catch (error) {
+      console.error("상품 상세 데이터 오류", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const getCommentsData = async () => {
+    try {
+      setIsLoading(true);
+      const data = await getComments({ productId });
+      setComments(data.list);
+    } catch (error) {
+      console.error("코멘트 데이터 오류", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (productId) {
+      getDetailData();
+      getCommentsData();
+    }
+  }, [productId]);
+
+  const handleEdit = (id, content) => {
+    setEditingCommentId(id);
+    setEditedContent(content);
+    setOpenMenuId(null); // 드롭다운 닫기
+  };
+
+  if (isLoading) {
+    return <p className="text-center animate-bounce ">로딩 중..</p>;
+  }
+
+  if (!item) {
+    return <p className="text-center animate-bounce">상품 정보가 없습니다.</p>;
+  }
+
+  return (
+    <div className="max-w-[120rem] mx-auto p-7">
+      <div className="flex flex-col md:flex-row w-full h-[25%] gap-10 mx-auto text-[#4B5563] border-b-gray-200 border-b pb-10">
+        <div className="w-full md:w-[35%] h-full">
+          <img
+            className="h-full aspect-1/1 rounded-3xl"
+            src={item.images[0] || noImage}
+          />
+        </div>
+        <div className="flex flex-col w-full md:w-[65%] justify-between">
+          <div className="flex flex-col gap-5 mb-5 text-[#1F2937]">
+            <div className="flex justify-between pr-5 ">
+              <h3 className="text-[2rem] font-[600]">{item.name}</h3>
+              <VerticalEllipsis />
+            </div>
+            <h1 className="text-[3.2rem] font-[600]">
+              {item.price.toLocaleString()}원
+            </h1>
+          </div>
+          <div className="flex flex-col gap-5 mb-5">
+            <h5 className="text-[1.4rem] font-[600]">상품 소개</h5>
+            <p className="text-[1.2rem] font-[400] h-[10vh]">
+              {item.description}
+            </p>
+          </div>
+          <div className="flex flex-col gap-5 mb-5">
+            <h5 className="text-[1.4rem] font-[600]">상품 태그</h5>
+            <div className="flex gap-2 flex-wrap mt-2">
+              {item.tags &&
+                item.tags.map((tag) => (

```

Comment:
💊 제안
item 의 기본값이 null이므로 아래처럼 optional chaining을 이용해서 작성하셔도 됩니다~
```suggestion
              {item?.tags?.map((tag) => (

```